markdown.dtx

```diff
@@ -33265,6 +33435,84 @@ end
   }%
 |endgroup
 %    \end{macrocode}
+% \begin{markdown}
+% The \mref{markinline} macro uses an implementation of the
+% \mref{markdownLuaExecute} macro to convert inline markdown to plain
+% \TeX{}.
+%
+% \end{markdown}
+%  \begin{macrocode}
+\begingroup
+%    \end{macrocode}
+% \begin{markdown}
+% Swap the category code of the backslash symbol and the pipe symbol, so that
+% we may use the backslash symbol freely inside the Lua code. Furthermore,
+% use the ampersand symbol to specify parameters.
+% \end{markdown}
+%  \begin{macrocode}
+  \catcode`|=0%
+  \catcode`\\=12%
+  \catcode`|&=6%
+  |gdef|markinline#1{%
```

Comment:
Just the Lua part would have been sufficient for this release. Buffering markdown text is tricky; using `\gdef\markinline#1{...}` will cause TeX to tokenize `#1`, which we do not want. Instead, we want to peek ahead using low-level commands and collect the individual characters after `\markinline`, so that we can pass them to Lua unaltered via a temporary file and `\markdownInput`. In an ongoing effort to reduce the technical debt, the implementation will be written in expl3, not plain TeX, similarly to commit c8be4a5a4dbfdcb114b3ab0cf5b8fcdb88e31591 from recent PR #351.

For these reasons, I suggest that we remove the TeX part of this PR and only keep the Lua part for this release. The TeX part can serve an an inspiration and a starting point for the implementation of `\markinline` in a future release.